File: ios/Classes/TrackManager.swift
import Foundation
import sariska

class TrackManager: NSObject {

    @objc func mute(_ params: NSDictionary) {
       for track in SariskaMediaTransport.getLocalTracks() {
          let track = track as! JitsiLocalTrack
          if (track.getId() == (params["trackId"] as! NSString))  {
              track.mute()
          }
      }
    }

    @objc func unmute(_ params: NSDictionary) {
       for track in SariskaMediaTransport.getLocalTracks() {
          let track = track as! JitsiLocalTrack
          if (track.getId() == (params["trackId"] as! NSString))  {
              track.unmute()
          }
       }
    }

    @objc func switchCamera(_ params: NSDictionary) {
       for track in SariskaMediaTransport.getLocalTracks() {
          let track = track as! JitsiLocalTrack
          if (track.getId() == (params["trackId"] as! NSString))  {
              track.switchCamera()
          }
       }
    }

    @objc func dispose(_ params: NSDictionary) {
       for track in SariskaMediaTransport.getLocalTracks() {
          let track = track as! JitsiLocalTrack
          if (track.getId() == (params["trackId"] as! NSString))  {
              track.dispose()
          }
       }
    }

    @objc func destroy() {

    }
}

File: android/src/main/kotlin/io/sariska/sariska_media_transport/TrackManager.kt
package io.sariska.sariska_media_transport
import io.sariska.sdk.SariskaMediaTransport


class TrackManager {
    fun mute(params: Map<String, *>) {
        for(track in SariskaMediaTransport.getLocalTracks()) {
            if (track.getId() == (params["trackId"] as? String))  {
                track.mute()
            }
        }
    }

    fun unmute(params: Map<String, *>) {
        for(track in SariskaMediaTransport.getLocalTracks()) {
            if (track.getId() == (params["trackId"] as? String))  {
                track.unmute()
            }
        }
    }

    fun switchCamera(params: Map<String, *>) {
        for(track in SariskaMediaTransport.getLocalTracks()) {
            if (track.getId() == (params["trackId"] as? String))  {
                track.switchCamera()
            }
        }
    }

    fun dispose(params: Map<String, *>) {
        for(track in SariskaMediaTransport.getLocalTracks()) {
            if (track.getId() == (params["trackId"] as? String))  {
                track.dispose()
            }
        }
    }

    fun destroy() {
    }
}
